### PR TITLE
Updates to 12-SNAPSHOT

### PR DESCRIPTION
**Pull request by michael-o**
* I have intentionally left distinct commits
* Some plugins where not updated due to Java 7 constraint
* Duplicate lines to ASF parent have been removed
* Obsolete config has been removed too

This change will require also changes to Core 5 and Client 5 POMs, but they need streamlining anyway. I have parts of the changes already locally and both build fine (master and 5.1.x.).

### CODE REVIEW

**Review by ok2c — APPROVED** [review]
@michael-o Looks good to me.

File: pom.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
    "License"); you may not use this file except in compliance
    with the License.  You may obtain a copy of the License at
 
-     http://www.apache.org/licenses/LICENSE-2.0
+     https://www.apache.org/licenses/LICENSE-2.0
```
Comment (garydgregory) [inline pom.xml]: This has come up in the past and I am not sure it is ok to change a single character in a license header. I think you'll want to check the LEGAL Jira project and/or the legal ML.

Comment (michael-o) [reply pom.xml]: Thanks for raising. See here https://issues.apache.org/jira/browse/LEGAL-416 and the result: http://www.apache.org/foundation/license-faq.html#Apply-My-Software The sample refers to https too. Is that OK? I don't mind to revert to HTTP if this causes a not permitted license change.

Comment (michael-o) [reply pom.xml]: Done with LEGAL-536.

**Comment by michael-o** [comment]
@garydgregory I have reverted the protocol change. See also: https://issues.apache.org/jira/projects/LEGAL/issues/LEGAL-457
Is there anything else or do you approve too?

**Comment by ok2c** [comment]
@michael-o Please go ahead, merge this change-set and put the new release on a lazy vote. @garydgregory will have a chance to object to any of those changes during the release vote if it comes to that.

**Comment by garydgregory** [comment]
I don't make legal policies at Apache.

**Comment by garydgregory** [comment]
https://www.apache.org/legal/src-headers.html

File: pom.xml [hunk 2]
```diff
@@ -22,22 +22,22 @@
    This software consists of voluntary contributions made by many
    individuals on behalf of the Apache Software Foundation.  For more
    information on the Apache Software Foundation, please see
-   <http://www.apache.org/>.
+   <https://www.apache.org/>.
```
Comment (ok2c) [inline pom.xml:25]: @michael-o let's revert changes to the ASL header and move on.

Comment (michael-o) [reply pom.xml:25]: Alright, will do.

**Comment by michael-o** [comment]
All in.

**Pull request closed without merging**